geozero/src/wkt/wkt_reader.rs

```diff
@@ -354,14 +341,75 @@ mod test {
         assert_eq!(str, &roundtripped);
     }
 
-    #[test]
-    fn geometry_collection_roundtrip() {
-        let str = "GEOMETRYCOLLECTION(POINT(40 10),LINESTRING(10 10,20 20,10 40),POLYGON((40 40,20 45,45 30,40 40)))";
-        let wkt = WktStr(str);
+    mod empties {
+        use super::*;
 
-        use crate::wkt::conversion::ToWkt;
-        let roundtripped = wkt.to_wkt().unwrap();
+        #[test]
+        fn empty_point() {
+            let wkt = WktStr("POINT EMPTY");
+            let actual = wkt.to_geo().unwrap_err();
+            assert!(matches!(actual, GeozeroError::Geometry(_)));
+        }
 
-        assert_eq!(str, &roundtripped);
+        #[test]
+        fn empty_point_roundtrip() {
+            let wkt = WktStr("POINT EMPTY");
+            let actual = wkt.to_wkt().unwrap();
+            assert_eq!("POINT EMPTY", &actual);
+        }
+
+        #[test]
+        fn geometry_collection_with_empty_point() {
+            let str = "GEOMETRYCOLLECTION(POINT(40 10),LINESTRING(10 10,20 20,10 40),POINT EMPTY)";
+            let wkt = WktStr(str);
+
+            use crate::wkt::conversion::ToWkt;
+            let roundtripped = wkt.to_wkt().unwrap();
+
+            assert_eq!(str, &roundtripped);
+        }
+
+        #[test]
+        fn empty_line_string() {
+            let wkt = WktStr("LINESTRING EMPTY");
+            let actual = wkt.to_geo().unwrap();
+            let expected: geo_types::Geometry<f64> = line_string![].into();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn empty_multi_line_string() {
+            let wkt = WktStr("MULTILINESTRING EMPTY");
+            let actual = wkt.to_geo().unwrap();
+            let expected: geo_types::Geometry<f64> = geo_types::MultiLineString(vec![]).into();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn empty_polygon() {
+            let wkt = WktStr("POLYGON EMPTY");
+            let actual = wkt.to_geo().unwrap();
+            let expected: geo_types::Geometry<f64> = polygon![].into();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn empty_multi_polygon() {
+            let wkt = WktStr("MULTIPOLYGON EMPTY");
+            let actual = wkt.to_geo().unwrap();
+            let expected: geo_types::Geometry<f64> = geo_types::MultiPolygon(vec![]).into();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        #[ignore = "geo::GeometryCollections is known to be broken https://github.com/georust/geozero/issues/11"]
+        fn empty_geometry_collection() {
+            let wkt = WktStr("GEOMETRYCOLLECTION EMPTY");
+            let actual = wkt.to_geo().unwrap();
+
+            let expected =
+                geo_types::Geometry::GeometryCollection(geo_types::GeometryCollection(vec![]));
```

Comment:
wildly, this currently returns `GeometryCollection(vec![point(x: 0.0, y: 0.0)])`. I *think* this will be fixed by https://github.com/georust/geozero/issues/11